File: custconnect_mobile/lib/presentation/screens/welcome_screen.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import 'package:lucide_icons/lucide_icons.dart';

class WelcomeScreen extends StatelessWidget {
  const WelcomeScreen({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    final size = MediaQuery.of(context).size;

    return Scaffold(
      body: Stack(
        children: [
          // Background Image (Solid color as fallback, but typically an Image.network here or AssetImage)
          Container(
            width: size.width,
            height: size.height,
            decoration: const BoxDecoration(
              image: DecorationImage(
                image: NetworkImage('https://images.unsplash.com/photo-1523050335456-e6cc8390b46d?q=80&w=1471&auto=format&fit=crop'),
                fit: BoxFit.cover,
              ),
            ),
          ),
          // Overlay
          Container(
            width: size.width,
            height: size.height,
            color: const Color(0xFF1A2744).withOpacity(0.85),
          ),
          SafeArea(
            child: Padding(
              padding: const EdgeInsets.all(30.0),
              child: Column(
                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                children: [
                  Row(
                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
                    children: [
                      Row(
                        children: [
                          Container(
                            width: 54,
                            height: 54,
                            decoration: BoxDecoration(
                              color: Colors.white.withOpacity(0.15),
                              borderRadius: BorderRadius.circular(18),
                              border: Border.all(color: Colors.white.withOpacity(0.2)),
                            ),
                            child: const Icon(LucideIcons.graduationCap, color: Colors.white, size: 32),
                          ),
                          const SizedBox(width: 15),
                          const Text(
                            'CustConnect',
                            style: TextStyle(fontSize: 30, fontWeight: FontWeight.w900, color: Colors.white, letterSpacing: -1),
                          ),
                        ],
                      ),
                      Container(
                        padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 6),
                        decoration: BoxDecoration(
                          color: theme.primaryColor,
                          borderRadius: BorderRadius.circular(10),
                        ),
                        child: const Text('v2.0 PREMIUM', style: TextStyle(color: Colors.white, fontSize: 10, fontWeight: FontWeight.w900)),
                      )
                    ],
                  ),
                  Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: const [
                      Text(
                        'Your Campus,\nPerfected.',
                        style: TextStyle(fontSize: 62, fontWeight: FontWeight.w900, color: Colors.white, height: 1.0, letterSpacing: -2.5),
                      ),
                      SizedBox(height: 25),
                      Text(
                        'The ultimate ecosystem for student life. Access your tools, social feed, and resources in one premium space.',
                        style: TextStyle(fontSize: 18, color: Colors.white70, height: 1.5, fontWeight: FontWeight.w500),
                      ),
                    ],
                  ),
                  Column(
                    children: [
                      ElevatedButton(
                        onPressed: () => context.push('/register'),
                        style: ElevatedButton.styleFrom(
                          backgroundColor: Colors.white,
                          foregroundColor: theme.colorScheme.secondary,
                          padding: const EdgeInsets.symmetric(vertical: 20),
                          shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(22)),
                          minimumSize: const Size(double.infinity, 60),
                        ),
                        child: Row(
                          mainAxisAlignment: MainAxisAlignment.center,
                          children: [
                            Text('Create Account', style: TextStyle(fontSize: 18, fontWeight: FontWeight.w900, color: theme.colorScheme.secondary)),
                            const SizedBox(width: 12),
                            Icon(LucideIcons.arrowRight, color: theme.colorScheme.secondary, size: 20),
                          ],
                        ),
                      ),
                      const SizedBox(height: 18),
                      TextButton(
                        onPressed: () => context.push('/login'),
                        child: RichText(
                          text: const TextSpan(
                            text: 'Already a member? ',
                            style: TextStyle(color: Colors.white70, fontSize: 16, fontWeight: FontWeight.w500),
                            children: [
                              TextSpan(text: 'Sign In', style: TextStyle(color: Colors.white, fontWeight: FontWeight.w900, decoration: TextDecoration.underline)),
                            ],
                          ),
                        ),
                      ),
                      TextButton(
                        onPressed: () => context.push('/forgot-password'),
                        child: const Text('Need technical support?', style: TextStyle(color: Colors.white54, fontSize: 13, fontWeight: FontWeight.w500)),
                      ),
                    ],
                  )
                ],
              ),
            ),
          ),
        ],
      ),
    );
  }
}

File: android_legacy_backup/src/screens/WelcomeScreen.js
import React from 'react';
import {
    View,
    Text,
    StyleSheet,
    TouchableOpacity,
    ImageBackground,
    Dimensions,
    SafeAreaView,
    StatusBar
} from 'react-native';
import { useTheme } from '../contexts/ThemeContext';
import { ArrowRight, GraduationCap } from 'lucide-react-native';

const { width, height } = Dimensions.get('window');
const BG_IMAGE = 'https://images.unsplash.com/photo-1523050335456-e6cc8390b46d?q=80&w=1471&auto=format&fit=crop';

export default function WelcomeScreen({ onStart, onForgotPassword, onLogin }) {
    const { colors, isDarkMode, shadows } = useTheme();

    return (
        <View style={styles.container}>
            <StatusBar barStyle="light-content" />
            <ImageBackground
                source={{ uri: BG_IMAGE }}
                style={styles.backgroundImage}
            >
                <View style={[styles.overlay, { backgroundColor: isDarkMode ? 'rgba(15, 23, 42, 0.9)' : 'rgba(26, 39, 68, 0.85)' }]} />

                <SafeAreaView style={styles.safeArea}>
                    <View style={styles.content}>
                        <View style={styles.header}>
                            <View style={styles.logoContainer}>
                                <View style={styles.iconCircle}>
                                    <GraduationCap size={32} color="white" />
                                </View>
                                <Text style={styles.brand}>CustConnect</Text>
                            </View>
                            <View style={[styles.badge, { backgroundColor: colors.primary }, shadows.sm]}>
                                <Text style={styles.badgeText}>v2.0 PREMIUM</Text>
                            </View>
                        </View>

                        <View style={styles.heroContainer}>
                            <Text style={styles.title}>Your Campus,{"\n"}Perfected.</Text>
                            <Text style={styles.subtitle}>
                                The ultimate ecosystem for student life. Access your tools, social feed, and resources in one premium space.
                            </Text>
                        </View>

                        <View style={styles.footer}>
                            <TouchableOpacity style={[styles.primaryButton, shadows.lg]} onPress={onStart} activeOpacity={0.8}>
                                <Text style={[styles.primaryButtonText, { color: colors.secondary }]}>Create Account</Text>
                                <ArrowRight size={20} color={colors.secondary} />
                            </TouchableOpacity>

                            <TouchableOpacity style={styles.secondaryButton} onPress={onLogin} activeOpacity={0.7}>
                                <Text style={styles.secondaryButtonText}>Already a member? </Text>
                                <Text style={styles.loginLink}>Sign In</Text>
                            </TouchableOpacity>

                            <TouchableOpacity style={styles.textButton} onPress={onForgotPassword}>
                                <Text style={styles.textButtonText}>Need technical support?</Text>
                            </TouchableOpacity>
                        </View>
                    </View>
                </SafeAreaView>
            </ImageBackground>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    backgroundImage: {
        width: width,
        height: height,
        flex: 1,
    },
    overlay: {
        ...StyleSheet.absoluteFillObject,
    },
    safeArea: {
        flex: 1,
    },
    content: {
        flex: 1,
        padding: 30,
        justifyContent: 'space-between',
    },
    header: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
        marginTop: 10,
    },
    logoContainer: {
        flexDirection: 'row',
        alignItems: 'center',
        gap: 15,
    },
    iconCircle: {
        width: 54,
        height: 54,
        borderRadius: 18,
        backgroundColor: 'rgba(255,255,255,0.15)',
        alignItems: 'center',
        justifyContent: 'center',
        borderWidth: 1,
        borderColor: 'rgba(255,255,255,0.2)',
    },
    brand: {
        fontSize: 30,
        fontWeight: '900',
        color: 'white',
        letterSpacing: -1,
    },
    badge: {
        paddingHorizontal: 12,
        paddingVertical: 6,
        borderRadius: 10,
    },
    badgeText: {
        fontSize: 10,
        fontWeight: '900',
        color: 'white',
        letterSpacing: 0.5,
    },
    heroContainer: {
        marginVertical: 40,
    },
    title: {
        fontSize: 62,
        fontWeight: '900',
        color: 'white',
        lineHeight: 64,
        letterSpacing: -2.5,
    },
    subtitle: {
        fontSize: 18,
        color: 'rgba(255,255,255,0.7)',
        marginTop: 25,
        lineHeight: 28,
        fontWeight: '500',
    },
    footer: {
        gap: 18,
        marginBottom: 10,
    },
    primaryButton: {
        backgroundColor: 'white',
        paddingVertical: 20,
        borderRadius: 22,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        gap: 12,
    },
    primaryButtonText: {
        fontSize: 18,
        fontWeight: '900',
    },
    secondaryButton: {
        flexDirection: 'row',
        justifyContent: 'center',
        paddingVertical: 10,
    },
    secondaryButtonText: {
        color: 'rgba(255,255,255,0.7)',
        fontSize: 16,
        fontWeight: '500',
    },
    loginLink: {
        color: 'white',
        fontSize: 16,
        fontWeight: '900',
        textDecorationLine: 'underline',
    },
    textButton: {
        alignItems: 'center',
        marginTop: 5,
    },
    textButtonText: {
        color: 'rgba(255,255,255,0.4)',
        fontSize: 13,
        fontWeight: '500',
    },
});
